Plugins/NativeCamera/Source/NativeCameraApple.mm

```diff
@@ -0,0 +1,149 @@
+            if (frontCamera) {
+                preferredPosition = AVCaptureDevicePositionFront;
+                preferredDeviceType = AVCaptureDeviceTypeBuiltInTrueDepthCamera;
+            } else {
+                preferredPosition = AVCaptureDevicePositionBack;
+                preferredDeviceType = AVCaptureDeviceTypeBuiltInDualCamera;
+            }
+
+            // Set camera capture device to default and the media type to video.
+            AVCaptureDevice* captureDevice = [AVCaptureDevice defaultDeviceWithDeviceType:preferredDeviceType mediaType:AVMediaTypeVideo position:preferredPosition];
+            if (!captureDevice) {
+                // If a rear dual camera is not available, default to the rear wide angle camera.
+                captureDevice = [AVCaptureDevice defaultDeviceWithDeviceType:AVCaptureDeviceTypeBuiltInWideAngleCamera mediaType:AVMediaTypeVideo position:AVCaptureDevicePositionBack];
+                
+                // In the event that the rear wide angle camera isn't available, default to the front wide angle camera.
+                if (!captureDevice) {
+                    captureDevice = [AVCaptureDevice defaultDeviceWithDeviceType:AVCaptureDeviceTypeBuiltInWideAngleCamera mediaType:AVMediaTypeVideo position:AVCaptureDevicePositionFront];
+                }
+            }
+#else
+            UNUSED(frontCamera);
+            AVCaptureDevice* captureDevice = [AVCaptureDevice defaultDeviceWithMediaType:AVMediaTypeVideo];
+#endif
+            // Set video capture input: If there a problem initialising the camera, it will give am error.
+            AVCaptureDeviceInput *input = [AVCaptureDeviceInput deviceInputWithDevice:captureDevice error:&error];
+
+            if (!input) {
+                NSLog(@"Error Getting Camera Input");
+                return;
+            }
+            // Adding input souce for capture session. i.e., Camera
+            [cameraInterfaceApple->avCaptureSession addInput:input];
+
+            dispatch_queue_t sampleBufferQueue = dispatch_queue_create("CameraMulticaster", DISPATCH_QUEUE_SERIAL);
+
+            AVCaptureVideoDataOutput * dataOutput = [[AVCaptureVideoDataOutput alloc] init];
+            [dataOutput setAlwaysDiscardsLateVideoFrames:YES];
+            [dataOutput setVideoSettings:@{(id)kCVPixelBufferPixelFormatTypeKey: @(kCVPixelFormatType_32BGRA)}];
+            [dataOutput setSampleBufferDelegate:cameraInterfaceApple->cameraTextureDelegate queue:sampleBufferQueue];
+
+            [cameraInterfaceApple->avCaptureSession addOutput:dataOutput];
+            [cameraInterfaceApple->avCaptureSession commitConfiguration];
+            [cameraInterfaceApple->avCaptureSession startRunning];
+        });
+        return cameraInterfaceApple;
+    }
+
+    void CameraInterfaceApple::UpdateCameraTexture(bgfx::TextureHandle textureHandle)
+    {
+        if (textureBGRA)
+        {
+            bgfx::overrideInternal(textureHandle, reinterpret_cast<uintptr_t>(textureBGRA));
```

Comment:
Is this going to work when the JavaScript thread is no longer the graphics thread?

Reply:
That's something I need to test after Gary's PR is in. I have big concerns about both ios and android platforms. UI access on ios and opengl context on android. I won't merge this PR before I know it works with new threading model.